build-system/app-index/test/helpers.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {expect} = require('chai');
+const {JSDOM} = require('jsdom');
+
+
+const parseHtmlChunk = htmlStr => {
+  const {body} = (new JSDOM(htmlStr)).window.document;
+  expect(body.children).to.have.length(1);
+  return body.firstElementChild;
+};
+
+
+const boundAttrRe = attr =>
+  new RegExp(`\\[${attr}\\]=(("[^"]+")|('[^']+')|([^\\s\\>]+))`);
+
+
+// JSDom doesn't parse attributes whose names don't follow the spec, so
+// our only way to test [attr] values is via regex.
+const getBoundAttr = (el, attr) => {
+  const match = el./*OK*/outerHTML.match(boundAttrRe(attr));
+  if (match) {
+    const [_, valuePart] = match;
+    if (valuePart.charAt(0) == '"' ||
+        valuePart.charAt(0) == '\'') {
+      return valuePart.substring(1, valuePart.length - 1);
+    }
+    return valuePart;
+  }
+}
+
+
+const expectValidAmphtml = (validator, string) => {
+  const {errors, status} = validator.validateString(string);
+  expect(errors).to.deep.equal([]);
```

Comment:
`expect(errors).to.be.empty`?

Reply:
One difference between using `.to.be.empty` is that the assertion output is different: 

```
AssertionError: expected [ Array(12) ] to be empty
    at expectValidAmphtml (build-system/app-index/test/helpers.js:51:23)
    ...
```

Whereas you get the actual comparison values when comparing to an empty array, so you get the validation errors for free:

```
AssertionError: expected [ Array(12) ] to deeply equal []
      + expected - actual

      -[
      -  {
      -    "category": "MANDATORY_AMP_TAG_MISSING_OR_INCORRECT"
      -    "code": "MANDATORY_ATTR_MISSING"
      -    "col": 0
      -    "line": 1
      -    "message": "The mandatory attribute '⚡' is missing in tag 'html'."                                                                                                                       [               -    "params": [
      -      "⚡"                                                                                                                                                                                    [               -      "html"
      -    ]
      -    "severity": "ERROR"
      -    "specUrl": "https://www.ampproject.org/docs/reference/spec#required-markup"
      -  }
      ...
```

Reply:
validateString returns the validationresult object, which has errors on it. note that errors will also include warnings. if just checking for validity then status would be better to check because if the given string only has warnings it's still valid and errors is not empty.